meta-balena-common/recipes-core/initrdscripts/files/resindataexpander

```diff
@@ -1,32 +1,40 @@
 #!/bin/sh
 
 FREESPACE_LIMIT=10
-datapart=$(readlink -f /dev/disk/by-label/resin-data)
-datadev=$(lsblk $datapart -n -o PKNAME)
 
 resindataexpander_enabled() {
     # On flasher avoid expanding partition
+    # shellcheck disable=SC2154
     if [ "$bootparam_flasher" = "true" ]; then
-        echo "[INFO] Flasher detected. Avoiding expand partition mechanism."
+        error "Flasher detected. Avoiding expand partition mechanism."
         return 1
     fi
-
-    for freespace in $(parted -m /dev/$datadev unit MiB print free | grep free | cut -d: -f4 | sed 's/MiB//g'); do
-        if [ $(echo $freespace \> $FREESPACE_LIMIT | bc -l) == "1" ]; then
-            return 0
-        fi
-    done
-    return 1
+    return 0
 }
 
 resindataexpander_run() {
-    info "resindataexpander: Expand extended partition... "
-    parted -s /dev/$datadev -- resizepart 4 -1s
-    info "resindataexpander: Finished expanding extended partition."
-    info "resindataexpander: Expand data partition... "
-    parted -s /dev/$datadev -- resizepart 6 -1s
-    info "resindataexpander: Finished expanding data partition."
+    datapart=$(get_state_path_from_label "resin-data")
+    if ! wait4file "$datapart" "300"; then
+	    error "Timeout while waiting for data partition to be detected. Data partition expansion will not be tried."
+	    return
```

Comment:
Should this return an error?

Reply:
See the comment in https://github.com/balena-os/meta-balena/pull/1840#discussion_r393536616
The ${module}_run function call does not check for a return value, so returning something is confusing.

Reply:
ack